### ISSUE TITLE
Test failures not showing in reporter

### ISSUE THREAD

**Issue by philcockfield** [Respondly/meteor-test-reporter#31]
The following commit:

https://github.com/Respondly/meteor-data/commit/c9b3e507591cd62615f961e9f8f9fd41b192e357

Produces this error condition:

![image](https://cloud.githubusercontent.com/assets/185555/6207176/44d2fb1e-b600-11e4-97c0-ba39fc92a4e6.png)

There are tests failing (see the `AssertionError`'s in the console), yet it is reporting 0 failed.

Curious to see within the commit reproduces this reliably for you. I was able to reliably put it in and out of this state before making the commit to see if I could share it with you @rissem

**Comment by rissem** [comment]
I'm seeing it too. Thanks for the clear repro.

**Comment by rissem** [comment]
So that was failing because the test was not asynchronous and and the failure happened after the test had already finished.

However, I'm seeing a similar error now where the header is correct, but the thumbs up is not replaced by the failing tests.

![screen shot 2015-03-03 at 7 48 21 pm](https://cloud.githubusercontent.com/assets/138143/6477923/f0b3d348-c1dd-11e4-8b02-7844671db53c.png)

**Comment by rissem** [comment]
To reproduce just make sure you have failures and start on the failed tab.

**Comment by rissem** [comment]
Now I'm unable to reproduce.

**Comment by tmeasday** [comment]
I suspect this issue may be at source similar to 

https://github.com/practicalmeteor/meteor-mocha/issues/11#issuecomment-154919538

@rissem for a start I think you need to catch exceptions thrown by `func()` here and pass them to mocha:  https://github.com/rissem/meteor-mocha-core/blob/master/server.js#L90 (essentially the same as this PR: https://github.com/practicalmeteor/meteor-mocha-core/compare/practicalmeteor...tmeasday:practicalmeteor)

It's weird because I'm not seeing the same issue with this reporter that I was with practicalmeteor's.

But I am seeing some strange stuff on:
  a) test changes (the reporter doesn't appear to re-run)
  b) browser refresh (the behaviour above)

Would you like me to log issues with simple reproductions?

**Comment by rissem** [comment]
@tmeasday yes, I'd appreciate that. Also, I'm not sure how much practicalmeteor/meteor-mocha-core has diverged from mike/meteor-mocha-core, but we really should try to unify them. mocha-core is meant to be the non-opinionated stuff like making mocha work w/ fibers.

**Comment by tmeasday** [comment]
@rissem see #44 #45 #46.

Also I couldn't figure out which reporter to use with `mike:mocha` on the commandline (spacejam or the like).

(Ideally both this reporter and `practicalmeteor:mocha` would both use the same core mocha package to register tests, and then run them in different ways I feel. Not sure how far away we are from that..)
